auto/otel

```diff
@@ -0,0 +1,27 @@
+
+# Copyright (C) NGINX, Inc.
+
+if [ $NXT_DEBUG = YES ]; then
+    NXT_OTEL_LIB_DIR=src/otel/
+    NXT_OTEL_LIB_LOC=src/otel/target/debug/libotel.a
+    NXT_OTEL_BUILD_FLAG=""
+else
+    NXT_OTEL_LIB_DIR=src/otel/
+    NXT_OTEL_LIB_LOC=src/otel/target/release/libotel.a
+    NXT_OTEL_BUILD_FLAG="--release"
+fi
+
+if [ $NXT_OTEL ]; then
+    NXT_OTEL_LIBS="-lssl -lcrypto"
+    if [ $(which pkgconf) ]; then
+        NXT_OTEL_LIBS="$(pkgconf openssl --cflags --libs)"
+    elif [ $(which pkg-config) ]; then
+        NXT_OTEL_LIBS="$(pkg-config openssl --cflags --libs)"
```

Comment:
We always just use `pkg-config`

However this whole check seems slightly iffy...

What if otel isn't installed?

The normal way for checking if something is available is (using auto/njs as an example)

```
nxt_feature="njs"
nxt_feature_name=NXT_HAVE_NJS
nxt_feature_run=no
nxt_feature_incs="$NXT_NJS_CFLAGS $NXT_NJS_AUX_CFLAGS"
nxt_feature_libs="$NXT_NJS_LIBS $NXT_NJS_AUX_LIBS"
nxt_feature_test="#include <njs.h>

                  #if NJS_VERSION_NUMBER < 0x000803
                  # error njs < 0.8.3 is not supported.
                  #endif

                  int main(void) {
                      njs_vm_t      *vm;
                      njs_vm_opt_t  opts;

                      njs_vm_opt_init(&opts);

                      vm = njs_vm_create(&opts);
                      if (vm == NULL)
                          return 1;
                      return 0;
                  }"
. auto/feature

if [ $nxt_found = no ]; then
    $echo
    $echo $0: error: no njs library \>= 0.8.3 found.
    $echo
    exit 1;
fi
```

Also note that `nxt_feature_name=NXT_HAVE_NJS` causes an entry to be set in `build/include/nxt_auto_config.h` if it's found.

Though I understand that may be tricky seeing as there is no C API for it...

Reply:
Hmm, so the rust side of things relies on OpenSSL...

We could make `--otel` automatically select `--openssl`, but then that would change the configuration of Unit...

We do need to do a proper check for working OpenSSL though... perhaps we just need to copy the relevant part from `auto/ssltls`

We should also check for `rustc` and `cargo` like we do in `auto/modules/wasm-wasi-component`